xds/src/test/java/io/grpc/xds/GrpcXdsClientImplDataTest.java

```diff
@@ -2342,6 +2344,113 @@ public void parseCluster_validateEdsSourceConfig() throws ResourceInvalidExcepti
         LoadBalancerRegistry.getDefaultRegistry());
   }
 
+  @Test
+  public void processCluster_parsesMetadata()
```

Comment:
Should there be another test that tests when both `filter_metatdata` and `typed_filter_metadata` fields are present in the metadata with the same key, then ONLY the `typed_filter_metadata` field will be parsed based on https://github.com/envoyproxy/envoy/blob/main/api/envoy/config/core/v3/base.proto#L217-L259?

This would test the different scenarios of `parseClusterMetadata(Metadata)` method above. In other words you would need to construct a test case where there is a key collision in `filter_metatdata` and `typed_filter_metadata`